include/proxsuite-nlp/linalg/bunchkaufman.hpp

```diff
@@ -341,18 +341,18 @@ bunch_kaufman_in_place_one_block(MatrixType &a, WType &w, IndicesType &pivots,
   }
 }
 
-template <typename MatrixType, typename VecType, typename IndicesType>
+template <typename MatrixType, typename VecType, typename IndicesType,
+          typename WorkspaceType>
 ComputationInfo bunch_kaufman_in_place(MatrixType &a, VecType &subdiag,
-                                       IndicesType &pivots,
+                                       IndicesType &pivots, WorkspaceType &w,
                                        Index &pivot_count) {
   Index blocksize = 32;
   Index n = a.rows();
 
   if (n <= blocksize) {
     blocksize = 0;
   }

```

Comment:
Why not giving blocksize as `bunch_kaufman_in_place` parameter ?

This avoid to hardcode 32 (already in BunchKaufman::BlockSize).

We can also use w.cols().

Reply:
You're right, we should use the workspace number of columns.